core/app/modules/kit-library/assets/js/pages/overview/overview-sidebar.js

```diff
@@ -39,10 +67,14 @@ export default function OverviewSidebar( props ) {
 					onChange={ setIsTagsCollapseOpen }
 					title={ __( 'TAGS', 'elementor' ) }
 					className="e-kit-library__item-sidebar-collapse-tags"
+					pageId="overview"
+					onClick={ ( val, title ) => {
+						onCollapseChange( val, title );
+					} }
 				>
 					<Grid container className="e-kit-library__item-sidebar-tags-container">
 						{ props.model.taxonomies.map( ( taxonomy ) => (
-							<OverviewTaxonomyBadge key={ taxonomy }>{ taxonomy }</OverviewTaxonomyBadge>
+							<OverviewTaxonomyBadge key={ taxonomy } kitName={ props.model.title } sidebarTagFilterEvent={ ( taxonomyText ) => sidebarTagFilterEvent( props.model.title, taxonomyText ) }>{ taxonomy }</OverviewTaxonomyBadge>
```

Comment:
Rename `sidebarTagFilterEvent` to be more generic

Reply:
onTagFilter?